frontend/src/pages/projects/screens/spawner/environmentVariables/AlertWarningText.tsx

```diff
@@ -0,0 +1,98 @@
+import { Alert, List, ListItem, Stack, StackItem } from '@patternfly/react-core';
+import React from 'react';
+
+type AlertWarningTextProps = {
+  deletedConfigMap: string[];
+  deletedSecret: string[];
+  loaded: boolean;
+};
+
+const AlertWarningText: React.FC<AlertWarningTextProps> = ({
+  deletedConfigMap,
+  deletedSecret,
+  loaded,
+}) => {
+  if (deletedConfigMap.length === 0 && deletedSecret.length === 0) {
+    return null;
+  }
+
+  let description;
+  const commonDescriptionText = `To remove the orphaned connection, save the workbench.`;
+
+  const formattedText = (items: string[], type: 'configMap' | 'secret') => {
+    if (items.length === 1) {
+      return (
+        <>
+          <strong>{items[0]}</strong> {type}
+        </>
+      );
+    }
+    const lastItem = items[items.length - 1];
+    const displayItems = items.slice(0, -1);
+    return (
+      <>
+        {displayItems.map((item, index) => (
+          <>
+            <strong key={index}>{item}</strong>
+            {index < displayItems.length - 1 ? ', ' : ' and '}
+          </>
+        ))}
+        <strong>{lastItem}</strong> {type}s
+      </>
+    );
+  };
+
+  if (deletedConfigMap.length > 0 && deletedSecret.length > 0) {
+    description = (
+      <Stack hasGutter>
+        <StackItem>
+          The following environment variables in this workbench cannot be found:
+        </StackItem>
+        <StackItem>
+          <List>
+            {[
+              formattedText(deletedConfigMap, 'configMap'),
+              formattedText(deletedSecret, 'secret'),
+            ].map((text, i) => (
+              <ListItem key={i}>{text}</ListItem>
+            ))}
+          </List>
+        </StackItem>
+        <StackItem>{commonDescriptionText}</StackItem>
+      </Stack>
+    );
+  } else if (deletedConfigMap.length > 0) {
+    description = (
+      <>
+        The {formattedText(deletedConfigMap, 'configMap')}
```

Comment:
This is called a couple times so might be better to hoist this as a variable of its own, along with the usage of `formattedText(deletedSecret, 'secret')`